internal/transform/offer_test.go

```diff
@@ -0,0 +1,167 @@
+
+	ingestio "github.com/stellar/go/exp/ingest/io"
+	"github.com/stellar/go/xdr"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestTransformOffer(t *testing.T) {
+	type transformTest struct {
+		input      ingestio.Change
+		wantOutput OfferOutput
+		wantErr    error
+	}
+	genericAccountID, err := xdr.NewAccountId(xdr.PublicKeyTypePublicKeyTypeEd25519, xdr.Uint256([32]byte{}))
+	assert.NoError(t, err)
+	genericAccountAddress, err := genericAccountID.GetAddress()
+	assert.NoError(t, err)
+
+	hardCodedInput, err := prepareHardcodedOfferTestInput()
+	assert.NoError(t, err)
+	hardCodedOutput := prepareHardcodedOfferTestOutput()
+
+	tests := []transformTest{
+		{
+			ingestio.Change{
+				Type: xdr.LedgerEntryTypeAccount,
+				Post: &xdr.LedgerEntry{
+					Data: xdr.LedgerEntryData{
+						Type: xdr.LedgerEntryTypeAccount,
+					},
+				},
+			},
+			OfferOutput{}, fmt.Errorf("Could not extract offer data from ledger entry; actual type is LedgerEntryTypeAccount"),
+		},
+		{
+			wrapOfferEntry(xdr.OfferEntry{
+				SellerId: genericAccountID,
+				OfferId:  -1,
+			}, 0),
+			OfferOutput{}, fmt.Errorf("OfferID is negative (-1) for offer from account: %s", genericAccountAddress),
+		},
+		{
+			wrapOfferEntry(xdr.OfferEntry{
+				SellerId: genericAccountID,
+				Amount:   -2,
+			}, 0),
+			OfferOutput{}, fmt.Errorf("Amount is negative (-2) for offer 0"),
+		},
+		{
+			wrapOfferEntry(xdr.OfferEntry{
+				SellerId: genericAccountID,
+				Price: xdr.Price{
+					N: -3,
+					D: 10,
+				},
+			}, 0),
+			OfferOutput{}, fmt.Errorf("Price numerator is negative (-3) for offer 0"),
+		},
+		{
+			wrapOfferEntry(xdr.OfferEntry{
+				SellerId: genericAccountID,
+				Price: xdr.Price{
+					N: 5,
+					D: -4,
+				},
+			}, 0),
+			OfferOutput{}, fmt.Errorf("Price denominator is negative (-4) for offer 0"),
+		},
+		{
+			wrapOfferEntry(xdr.OfferEntry{
+				SellerId: genericAccountID,
+				Price: xdr.Price{
+					N: 5,
+					D: 0,
+				},
+			}, 0),
+			OfferOutput{}, fmt.Errorf("Price denominator is 0 for offer 0"),
+		},
+		{
+			hardCodedInput,
+			hardCodedOutput, nil,
+		},
+	}
+
+	for _, test := range tests {
+		actualOutput, actualError := TransformOffer(test.input)
+		assert.Equal(t, test.wantErr, actualError)
+		assert.Equal(t, test.wantOutput, actualOutput)
+	}
+}
+
+func wrapOfferEntry(offerEntry xdr.OfferEntry, lastModified int) ingestio.Change {
+	return ingestio.Change{
+		Type: xdr.LedgerEntryTypeOffer,
+		Pre:  nil,
+		Post: &xdr.LedgerEntry{
+			LastModifiedLedgerSeq: xdr.Uint32(lastModified),
+			Data: xdr.LedgerEntryData{
+				Type:  xdr.LedgerEntryTypeOffer,
+				Offer: &offerEntry,
+			},
+		},
+	}
+}
+
+func prepareHardcodedOfferTestInput() (ledgerChange ingestio.Change, err error) {
+	hardCodedAccountID, err := xdr.NewAccountId(xdr.PublicKeyTypePublicKeyTypeEd25519, xdr.Uint256([32]byte{9, 13, 119, 101, 93, 2, 49, 146, 108, 232, 96, 108, 62, 49, 254, 143, 121, 165, 44, 237, 34, 197, 125, 11, 184, 24, 88, 236, 241, 192, 185, 158}))
```

Comment:
I feel like you've been calling this `hardCodedAccountID` value (and `hardCodedAssetIssuer`) in several different tests. Might be worth refactoring this as a `var` in one testing file, so you can just reference it across tests (since they're all in the same package, this should be easy).